setup.py

```diff
@@ -17,6 +17,38 @@
 with open(path_version) as f:
     version = f.read().strip()
 
+extra_requirements = {
+    'docs': [
+        # pin sphinx to match what RTD uses:
+        # https://github.com/readthedocs/readthedocs.org/blob/ecac31de54bbb2c100f933e86eb22b0f4389ba84/requirements/pip.txt#L16
+        'sphinx',
+        'sphinx_rtd_theme',
+        'sphinx-tabs==3.2.0',
+        'sphinx-toolbox==2.15.2',
+        'sphinx-jsonschema~=1.16',
+        'pypandoc',
+    ],
+    'tests': [
+        'pytest~=6.2',
+        'pytest-cases~=3.6.8',
+        'pytest-cov',
+        'pytest-ordering~=0.6',
+        'pytest-console-scripts~=1.1',
+        'coverage',
+        'coveralls',
+    ],
+    'contrib': [
+        'pre-commit>=2.10.1',
+        'flake8',    
+    ]
+}
+
+extra_requirements['dev'] = [
+    requirements,
+    extra_requirements['docs'],
+    extra_requirements['tests'],
+    extra_requirements['contrib'],
+    ]
```

Comment:
Ohhhhh that's smart. I have seen people begging pip/setuptools to add this feature.

I'm surprised this worked -- I would expect you to have to flatten these lists into a single list -- but CI passed so it must have worked?

**EDIT**: yes, [it did](https://github.com/ivadomed/ivadomed/runs/6636879430?check_suite_focus=true#step:4:88). Neat!

![Screenshot 2022-05-28 at 12-47-30 Modernize installing the CUDA 11 pytorch package · ivadomed_ivadomed@39a4d1f](https://user-images.githubusercontent.com/987487/170834874-bc204c3a-7807-48b9-96f2-60a5c656ecbe.png)

![Screenshot 2022-05-28 at 12-47-21 Modernize installing the CUDA 11 pytorch package · ivadomed_ivadomed@39a4d1f](https://user-images.githubusercontent.com/987487/170834878-935a1569-9038-4b06-acdd-e468f59561f2.png)

Cool, I'm glad I learned about this from you. I'll definitely be using it in the future.

Reply:
I have been observing this from quite some time 😅